mteb/tasks/Retrieval/multilingual/MLQARetrieval.py

```diff
@@ -0,0 +1,162 @@
+import datasets
+
+from mteb.abstasks import AbsTaskRetrieval, CrosslingualTask, TaskMetadata
+
+_LANGUAGES = {
+    "mlqa.ar.ar": ["ara-Arab", "ara-Arab"],
+    "mlqa.ar.de": ["ara-Arab", "deu-Latn"],
+    "mlqa.ar.en": ["ara-Arab", "eng-Latn"],
+    "mlqa.ar.es": ["ara-Arab", "spa-Latn"],
+    "mlqa.ar.hi": ["ara-Arab", "hin-Deva"],
+    "mlqa.ar.vi": ["ara-Arab", "vie-Latn"],
+    "mlqa.ar.zh": ["ara-Arab", "cmn-Hans"],
+    "mlqa.de.ar": ["deu-Latn", "ara-Arab"],
+    "mlqa.de.de": ["deu-Latn", "deu-Latn"],
+    "mlqa.de.en": ["deu-Latn", "eng-Latn"],
+    "mlqa.de.es": ["deu-Latn", "spa-Latn"],
+    "mlqa.de.hi": ["deu-Latn", "hin-Deva"],
+    "mlqa.de.vi": ["deu-Latn", "vie-Latn"],
+    "mlqa.de.zh": ["deu-Latn", "cmn-Hans"],
+    "mlqa.en.ar": ["eng-Latn", "ara-Arab"],
+    "mlqa.en.de": ["eng-Latn", "deu-Latn"],
+    "mlqa.en.en": ["eng-Latn", "eng-Latn"],
+    "mlqa.en.es": ["eng-Latn", "spa-Latn"],
+    "mlqa.en.hi": ["eng-Latn", "hin-Deva"],
+    "mlqa.en.vi": ["eng-Latn", "vie-Latn"],
+    "mlqa.en.zh": ["eng-Latn", "cmn-Hans"],
+    "mlqa.es.ar": ["spa-Latn", "ara-Arab"],
+    "mlqa.es.de": ["spa-Latn", "deu-Latn"],
+    "mlqa.es.en": ["spa-Latn", "eng-Latn"],
+    "mlqa.es.es": ["spa-Latn", "spa-Latn"],
+    "mlqa.es.hi": ["spa-Latn", "hin-Deva"],
+    "mlqa.es.vi": ["spa-Latn", "vie-Latn"],
+    "mlqa.es.zh": ["spa-Latn", "cln-Hans"],
+    "mlqa.hi.ar": ["hin-Deva", "ara-Arab"],
+    "mlqa.hi.de": ["hin-Deva", "deu-Latn"],
+    "mlqa.hi.en": ["hin-Deva", "eng-Latn"],
+    "mlqa.hi.es": ["hin-Deva", "spa-Latn"],
+    "mlqa.hi.hi": ["hin-Deva", "hin-Deva"],
+    "mlqa.hi.vi": ["hin-Deva", "vie-Latn"],
+    "mlqa.hi.zh": ["hin-Deva", "cmn-Hans"],
+    "mlqa.vi.ar": ["vie-Latn", "ara-Arab"],
+    "mlqa.vi.de": ["vie-Latn", "deu-Latn"],
+    "mlqa.vi.en": ["vie-Latn", "eng-Latn"],
+    "mlqa.vi.es": ["vie-Latn", "spa-Latn"],
+    "mlqa.vi.hi": ["vie-Latn", "hin-Deva"],
+    "mlqa.vi.vi": ["vie-Latn", "vie-Latn"],
+    "mlqa.vi.zh": ["vie-Latn", "cmn-Hans"],
+    "mlqa.zh.ar": ["cmn-Hans", "ara-Arab"],
+    "mlqa.zh.de": ["cmn-Hans", "deu-Latn"],
+    "mlqa.zh.en": ["cmn-Hans", "eng-Latn"],
+    "mlqa.zh.es": ["cmn-Hans", "spa-Latn"],
+    "mlqa.zh.hi": ["cmn-Hans", "hin-Deva"],
+    "mlqa.zh.vi": ["cmn-Hans", "vie-Latn"],
+    "mlqa.zh.zh": ["cmn-Hans", "cmn-Hans"],
+}
+
+_ISO6393_to_ISO6391 = {
+    "ara-Arab": "ar",
+    "deu-Latn": "de",
+    "eng-Latn": "en",
+    "spa-Latn": "es",
+    "hin-Deva": "hi",
+    "vie-Latn": "vi",
+    "zho-Hans": "zh",
+}
+
+
+def extend_lang_pairs() -> dict[str, list[str]]:
+    # add all possible language pairs
+    hf_lang_subset2isolang = {}
+    for x in _ISO6393_to_ISO6391.keys():
+        for y in _ISO6393_to_ISO6391.keys():
+            if x != y:
+                pair = f"{x}_{y}"
```

Comment:
Would each pair be query in xx -> corpus in yy?

Reply:
The dataset has a subset for each language pair xx->yy, I'd prefer not mixing them to avoid errors.

So what I did for now is use each subset: xx->yy on it's own. Since all texts are machine translated. 

You can see the subsets in the _LANGUAGES variable. Although, I'll change the way I'm handling these pairs because this is not very clear to understand.

Reply:
Oh yes I'll add to the discussion. I think whichever convention we end up choosing, we just need to write it down somewhere. I tend to forget 🙈 , so it'll be helpful to be able to check from time to time.